### DISCUSSION TITLE
regarding having to do audit where i only got etherscan links

### DISCUSSION THREAD

**Discussion by Jaideep555** [Cyfrin/security-and-auditing-full-course-s23#231] [Q&A]
Hello. I somehow got a assignment for a job basically but in that assignment all i got were etherscan links. I am confused how to proceed as i have having issues compiling it as the contracts depend on each other and compiling causes issue like import  and i have to edit all. Can anyone give me some idea how to proceed? 

Also sorry in advance as i am quite new to this and have only 3 days till deadline is over,

Thanks a lot

**Comment by EngrPips** [comment]
Hello @Jaideep555, Patrick advised us not to audit any protocol that gives us a link to some etherscan codebase. Did you the `Security and Auditing` course yet?

**Reply by Jaideep555** [reply]
hello @EngrPips , yes I know about that as I am also studying the course right now as well. But he also mentioned that some times we get just the etherscan link, so that is why i want to know

**Reply by EngrPips** [reply]
You can try re-creating the codebase locally if you want.

**Reply by Jaideep555** [reply]
thx a lot!. That is what i am actually trying but it's causing lot of errors at compiling and consuming too much time. Is there any alternative?

**Reply by EngrPips** [reply]
I don't know of any.

**Reply by Jaideep555** [reply]
ok, thx a lot though

**Reply by EngrPips** [reply]
You are welcome
